aptos-indexer-processor-example/src/utils/starting_version.rs

```diff
@@ -2,67 +2,252 @@ use super::database::ArcDbPool;
+        let mut db = PostgresTestDatabase::new();
+        let backfill_alias = "backfill_processor".to_string();
+        db.setup().await.unwrap();
+        let indexer_processor_config = create_indexer_config(
+            db.get_db_url(),
+            Some(BackfillConfig {
+                backfill_alias: backfill_alias.clone(),
+            }),
+        );
+        let conn_pool = new_db_pool(
+            db.get_db_url().as_str(),
+            Some(indexer_processor_config.db_config.db_pool_size),
+        )
+        .await
+        .expect("Failed to create connection pool");
+        run_migrations(db.get_db_url(), conn_pool.clone()).await;
+        diesel::insert_into(backfill_processor_status::table)
+            .values(BackfillProcessorStatus {
+                backfill_alias: backfill_alias.clone(),
+                backfill_status: BackfillStatus::Complete,
+                last_success_version: 10,
+                last_transaction_timestamp: None,
+                backfill_start_version: 0,
+                backfill_end_version: 10,
+            })
+            .execute(&mut conn_pool.clone().get().await.unwrap())
+            .await
+            .expect("Failed to insert processor status");
+
+        let starting_version = get_starting_version(&indexer_processor_config, conn_pool)
+            .await
+            .unwrap();
+
+        assert_eq!(starting_version, 0);
+    }
+
+    #[tokio::test]
+    async fn test_backfill_get_starting_version_with_inprogress_checkpoint() {
+        let mut db = PostgresTestDatabase::new();
+        let backfill_alias = "backfill_processor".to_string();
+        db.setup().await.unwrap();
+        let indexer_processor_config = create_indexer_config(
+            db.get_db_url(),
+            Some(BackfillConfig {
+                backfill_alias: backfill_alias.clone(),
+            }),
+        );
+        let conn_pool = new_db_pool(
+            db.get_db_url().as_str(),
+            Some(indexer_processor_config.db_config.db_pool_size),
+        )
+        .await
+        .expect("Failed to create connection pool");
+        run_migrations(db.get_db_url(), conn_pool.clone()).await;
+        diesel::insert_into(backfill_processor_status::table)
+            .values(BackfillProcessorStatus {
+                backfill_alias: backfill_alias.clone(),
+                backfill_status: BackfillStatus::InProgress,
+                last_success_version: 10,
+                last_transaction_timestamp: None,
+                backfill_start_version: 0,
+                backfill_end_version: 10,
+            })
+            .execute(&mut conn_pool.clone().get().await.unwrap())
+            .await
+            .expect("Failed to insert processor status");
+
+        let starting_version = get_starting_version(&indexer_processor_config, conn_pool)
+            .await
+            .unwrap();
+
+        assert_eq!(starting_version, 11);
     }
```

Comment:
nit: should we consider adding a test case where the configuration has a starting_version explicitly set?